.github/workflows/python-build-test.yml

```diff
@@ -41,3 +41,20 @@ jobs:
       run: |
         inv build
         inv build.mkdocs
+  build-rust:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out from Git
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ".[build,test,development,documentation]"
+      - name: Install Rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Rust Tests
+        run: cd rust && cargo build --all-targets && cargo test
```

Comment:
The rust toolchain action is slightly confused at teardown because the Cargo.toml file isn't in the root directory, but otherwise it seems to work. 

Should this use the python matrix to test with all supported versions?

Reply:
I went ahead and matrix-ified it.